hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/entity/impl/ReactiveAbstractEntityPersister.java

```diff
@@ -397,14 +397,14 @@ void preInsertInMemoryValueGeneration(Object[] fields, Object object,
 
 		ReactiveConnection connection = getReactiveConnection( session );
 		CompletionStage<? extends Serializable> generatedIdStage;
-		if ( getFactory().getSessionFactoryOptions().isGetGeneratedKeysEnabled() ) {
+//		if ( getFactory().getSessionFactoryOptions().isGetGeneratedKeysEnabled() ) {
 			generatedIdStage = connection.insertAndSelectIdentifier( checkSql( sql ), params );
-		}
-		else {
-			//use an extra round trip to fetch the id
-			generatedIdStage =  connection.update( sql, params )
-					.thenCompose( v -> connection.selectIdentifier( delegate().getIdentitySelectString(), EMPTY_OBJECT_ARRAY ) );
-		}
+//		}
+//		else {
+//			//use an extra round trip to fetch the id
+//			generatedIdStage =  connection.update( sql, params )
+//					.thenCompose( v -> connection.selectIdentifier( delegate().getIdentitySelectString(), EMPTY_OBJECT_ARRAY ) );
+//		}
```

Comment:
What's the reason for this change?

Reply:
When we use the extra trip the the identity generation test fails.

But why we do it? All the other databases support the single query insert select or similar

Reply:
OK, I thought we needed that on MySQL, but if we don't that's fine.

Reply:
Oh now I just read your very clear commit message and now I understand. I had actually misinterpreted the change. So actually this was a good catch! 👍